### DISCUSSION TITLE
External IP stacks when ISP IP changes

### DISCUSSION THREAD

**Discussion by dramirezRT** [RunOnFlux/fluxnode-multitool#137] [General]
Hi,
when the ISP changes the external IP the cron job updates the new address to the relevant interface. Although, it only adds and does no "housekeeping" like removing the old IP and so on. 

In my case IP changes once per day, and as you can see there are two external IPs assigned to the interface, the "old" one and the new one.

![image psd(1)](https://github.com/RunOnFlux/fluxnode-multitool/assets/8486626/fca5dfa2-ad8d-4326-a66a-79f32acf9147)

In my case, per day the IPs stacked do not go over two, although there may be more at any time.
Should we consider cleaning those IPs?

**Comment by XK4MiLX** [comment]
Try multitoolbox CleanIP -> Option 10 and re-create script

**Reply by dramirezRT** [reply]
Do you mean only running the multitoolbox with option 10?

**Reply by XK4MiLX** [reply]
type "multitoolbox CleanIP"

**Reply by dramirezRT** [reply]
I believe this works!
Similar approach, but better maybe, from what I was suggesting here: https://github.com/RunOnFlux/fluxnode-multitool/pull/86

**Reply by dramirezRT** [reply]
Just as a heads up I added the `--retry 5` to the curl command for the confirmed_ip since I saw some cases were it wouldn't work the first time, this way it wouldn't "skip" the IP update and wait for the next cronjob occurrence.

I also added `[[ "${WANIP}/32" == "${current_reg_ip}" ]] && exit 0` because once the IP is updated, due to the external IP change, it'd take a while for the API `http://localhost:$api_port/flux/info` to update to the new IP, this way it avoids constantly removing and adding the new IP untill it is confirmed by the API
